game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -493,8 +493,12 @@ private Matches() {}
             && !unitCanBeCapturedOnEnteringToInThisTerritory(player, terr, data).test(unit);
   }
 
-  public static Predicate<Unit> unitIsSuicide() {
-    return unit -> UnitAttachment.get(unit.getType()).getIsSuicide();
+  public static Predicate<Unit> unitIsSuicideOnAttack() {
+    return unit -> UnitAttachment.get(unit.getType()).getIsSuicideOnAttack();
+  }
+
+  public static Predicate<Unit> unitIsSuicideOnDefense() {
+    return unit -> UnitAttachment.get(unit.getType()).getIsSuicideOnDefense();
```

Comment:
Side-note, do you recall the rationale to not attach this method as a property of `Unit`? I suspect it would be pretty clearly better encapsulation.

Reply:
All the unit properties are part of UnitAttachment. This is really just the overall structure of the XML and code base.